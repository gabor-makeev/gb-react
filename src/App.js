import style from './App.module.scss'
import {Message} from "./components/Message/Message"

function App() {
  const messageContent = 'The content of a message'

  return (
    <div className={style.App}>
      <Message content={messageContent} />
    </div>
  );
}

export default App
