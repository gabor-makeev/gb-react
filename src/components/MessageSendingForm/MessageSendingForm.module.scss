.message-form {
  //Custom styles should go here
}