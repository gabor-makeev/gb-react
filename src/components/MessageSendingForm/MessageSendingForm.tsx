import style from './MessageSendingForm.module.scss';
import { STYLES } from '../../constants';
import React, { FC, memo, useState } from 'react';

import { Button } from './components/Button/Button';
import { Input } from './components/Input/Input';
import { Container } from '@mui/material';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { addMessage } from '../../store/messages/actions';

interface MessageFormProps {
  border?: string;
  borderRadius?: string;
}

export const MessageSendingForm: FC<MessageFormProps> = memo(
  ({ border = STYLES.border, borderRadius = STYLES.borderRadius }) => {
    const [inputValue, setInputValue] = useState('');
    const dispatch = useDispatch();
    const { chatId } = useParams();

    const formContainerStyle = {
      border: border,
      borderRadius: borderRadius,
      backgroundColor: '#CCCCCC',
      width: '300px',
      boxShadow: 'rgba(0, 0, 0, 0.35) 0 5px 15px',
      padding: '20px',
      display: 'flex',
      gap: '10px',
      justifyContent: 'space-between',
      margin: '0',
    };

    const sendMessage = (e: React.FormEvent<HTMLFormElement>) => {
      e.preventDefault();

      if (chatId) {
        dispatch(addMessage(chatId, inputValue));
      }

      setInputValue('');
    };

    return (
      <form
        onSubmit={sendMessage}
        className={style['message-form']}
        style={{ width: '100%' }}
      >
        <Container sx={formContainerStyle}>
          <Input
            disabled={!chatId}
            value={inputValue}
            setValue={setInputValue}
          />
          <Button disabled={!inputValue} />
        </Container>
      </form>
    );
  }
);
