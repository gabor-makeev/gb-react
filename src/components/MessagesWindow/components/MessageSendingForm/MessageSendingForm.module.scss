@import "src/styles/variables/colors";

.message-form {
  max-width: 762px;
  padding-top: 10px;
  height: 79px;
  flex-shrink: 0;
  width: 100%;
  display: flex;
  gap: 24px;
  justify-content: space-between;
  margin: 0;
}

.border-normal {
  border: 1px solid $policeBlue;
}

.border-bold {
  border: 5px solid $policeBlue;
}

.angle-roundish {
  border-radius: 15px;
}

.angle-square {
  border-radius: unset;
}