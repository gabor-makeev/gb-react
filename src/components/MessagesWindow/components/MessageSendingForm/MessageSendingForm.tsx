import style from './MessageSendingForm.module.scss';
import React, { FC, memo } from 'react';

import { Button } from './components/Button/Button';
import { Input } from './components/Input/Input';
import classNames from 'classnames';

interface MessageFormProps {
  inputValue?: string;
  setInputValue: (value: string) => void;
  isInputDisabled?: boolean;
  onSendMessage: (e: React.FormEvent<HTMLFormElement>) => void;
  borderType?: string;
  angleType?: string;
}

export const MessageSendingForm: FC<MessageFormProps> = memo(
  ({
    inputValue = '',
    setInputValue,
    onSendMessage,
    isInputDisabled = false,
    borderType,
    angleType,
  }) => {
    const classes = classNames(style['message-form'], {
      [style[`border-${borderType}`]]: borderType,
      [style[`angle-${angleType}`]]: angleType,
    });

    return (
      <form onSubmit={(e) => onSendMessage(e)} className={classes}>
        <Input
          disabled={isInputDisabled}
          value={inputValue}
          setValue={setInputValue}
        />
        <Button disabled={!inputValue} />
      </form>
    );
  }
);
