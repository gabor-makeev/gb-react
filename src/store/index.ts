import { combineReducers } from 'redux';
import { profileReducer } from './profile/slice';
import { chatsReducer } from './chats/slice';
import createSagaMiddleware from 'redux-saga';
import mySaga from './sagas';
import { configureStore } from '@reduxjs/toolkit';
import { articlesReducer } from 'store/articles/slice';

export type StoreState = ReturnType<typeof rootReducer>;

const rootReducer = combineReducers({
  profile: profileReducer,
  chats: chatsReducer,
  articles: articlesReducer,
});

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: rootReducer,
  devTools: process.env.NODE_ENV !== 'production',
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(mySaga);
