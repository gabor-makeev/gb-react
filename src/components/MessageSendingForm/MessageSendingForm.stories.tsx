import { MessageSendingForm } from './MessageSendingForm';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Provider } from 'react-redux';
import { store } from '../../store';

export default {
  title: 'MessageForm',
  component: MessageSendingForm,
} as ComponentMeta<typeof MessageSendingForm>;

const Template: ComponentStory<typeof MessageSendingForm> = (args) => (
  <Provider store={store}>
    <MessageSendingForm {...args} />
  </Provider>
);

export const Primary = Template.bind({});

export const BoldBorder = Template.bind({});
BoldBorder.args = {
  border: '5px solid black',
};

export const SquareBorders = Template.bind({});
SquareBorders.args = {
  borderRadius: 'unset',
};
