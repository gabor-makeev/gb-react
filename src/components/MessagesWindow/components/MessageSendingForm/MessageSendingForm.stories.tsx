import { MessageSendingForm } from './MessageSendingForm';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Provider } from 'react-redux';
import { store } from '../../../../store';

export default {
  title: 'MessageSendingForm',
  component: MessageSendingForm,
} as ComponentMeta<typeof MessageSendingForm>;

const Template: ComponentStory<typeof MessageSendingForm> = (args) => (
  <Provider store={store}>
    <MessageSendingForm {...args} />
  </Provider>
);

export const Primary = Template.bind({});

export const BoldBorder = Template.bind({});
BoldBorder.args = {
  borderType: 'bold',
};

export const SquareBorders = Template.bind({});
SquareBorders.args = {
  angleType: 'square',
};
