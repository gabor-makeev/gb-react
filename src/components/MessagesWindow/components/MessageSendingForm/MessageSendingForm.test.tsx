import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MessageSendingForm } from './MessageSendingForm';
import { Provider } from 'react-redux';
import { store } from 'src/store';

describe('MessageForm', () => {
  it('should render', () => {
    const mockHandler = jest.fn();
    render(
      <Provider store={store}>
        <MessageSendingForm
          onSendMessage={mockHandler}
          setInputValue={mockHandler}
        />
      </Provider>
    );
  });

  it('should have the button disabled when no input entered', () => {
    const mockHandler = jest.fn();
    render(
      <Provider store={store}>
        <MessageSendingForm
          onSendMessage={mockHandler}
          setInputValue={mockHandler}
        />
      </Provider>
    );

    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('should render with snapshot', () => {
    const mockHandler = jest.fn();
    const { asFragment } = render(
      <Provider store={store}>
        <MessageSendingForm
          onSendMessage={mockHandler}
          setInputValue={mockHandler}
          borderType={'normal'}
          angleType={'roundish'}
        />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
