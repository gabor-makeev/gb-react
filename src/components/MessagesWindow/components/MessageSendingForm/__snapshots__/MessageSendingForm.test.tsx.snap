// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should render with snapshot 1`] = `
<DocumentFragment>
  <form
    class="message-form border-normal angle-roundish"
  >
    <div
      class="MuiFormControl-root MuiTextField-root input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r2:"
        id=":r2:-label"
      >
        Message
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r2:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Message
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled button css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="message-send-button"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      Send
    </button>
  </form>
</DocumentFragment>
`;
